### PR TITLE
feat(builtin): generate `@npm//foo__all_files` filegroup that includes all files in the npm package

### PR DESCRIPTION
**Pull request by gregmagolan**
Including files with spaces & unicode characters & files that have been filtered out by the `included_files` attribute. This filegroup cannot be used in runfiles because of Bazel issue https://github.com/bazelbuild/bazel/issues/4327, but it can be used for other purposes such as the srcs input to a pkg_tar for generating a tar of an npm package pulled down with yarn_install or npm_install.

Needed for https://github.com/angular/angular/pull/33927 where pkg_tar is used to make a tar.gz of node_modules/puppeteer and on OSX the downloaded Chrome libs have spaces in their names:

```
def npm_package_archives():
    """Function to generate pkg_tar definitions for WORKSPACE yarn_install manual_build_file_contents"""
    npm_packages_to_archive = NPM_PACKAGE_ARCHIVES
    result = """load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
"""
    for name in npm_packages_to_archive:
        label_name = _npm_package_archive_label(name)
        last_segment_name = name if name.find("/") == -1 else name.split("/")[-1]
        result += """pkg_tar(
    name = "{label_name}",
    # use //{name}:{last_segment_name}__all_files as this includes **all** of the npm package
    # files including files with spaces and unicode chars which are not valid in Bazel runfiles.
    srcs = ["//{name}:{last_segment_name}__all_files"],
    extension = "tar.gz",
    strip_prefix = "./node_modules/{name}",
    # should not be build unless it is a dependency of another rule
    tags = ["manual"],
)
""".format(name = name, label_name = label_name, last_segment_name = last_segment_name)
    return result
```

See https://github.com/angular/angular/pull/33927#issuecomment-581982013 for more details.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/npm_install/test/golden/rxjs/BUILD.bazel.golden [hunk 1]
```diff
@@ -13,7 +13,6 @@ filegroup(
         "//:node_modules/rxjs/InnerSubscriber.d.ts",
         "//:node_modules/rxjs/InnerSubscriber.js",
         "//:node_modules/rxjs/InnerSubscriber.js.map",
-        "//:node_modules/rxjs/LICENSE.txt",
```
Comment (alexeagle) [inline internal/npm_install/test/golden/rxjs/BUILD.bazel.golden]: I guess it's technically a breaking change if users referred to one of these target directly?

File: internal/npm_install/generate_build_file.ts [hunk 2]
```diff
@@ -854,99 +849,116 @@ function findFile(pkg: Dep, m: string) {
...
+  // See https://github.com/bazelbuild/bazel/issues/4327.
+  const notPkgFiles = pkg._files.filter(
+      (f: string) => !f.startsWith('node_modules/') && !includedRunfiles.includes(f));
+  const notPkgFilesStarlark = notPkgFiles.length ? starlarkFiles('srcs', notPkgFiles) : '';
+
+  // If the package is in the Angular package format returns list
+  // of package files that end with `.umd.js`, `.ngfactory.js` and `.ngsummary.js`.
+  // TODO(gmagolan): add UMD & AMD scripts to scripts even if not an APF package _but_ only if they
+  // are named?
+  const namedSources = isNgApfPackage(pkg) ?
```
Comment (alexeagle) [inline internal/npm_install/generate_build_file.ts]: so many changes in this file - I'm assuming it's just a refactoring moving lines around, and not this many lines edited?

File: internal/npm_install/test/golden/rxjs/BUILD.bazel.golden (hunk 1 above)
Comment (gregmagolan) [reply internal/npm_install/test/golden/rxjs/BUILD.bazel.golden]: Nope. No breaking change here. This is just testing that  the `include_files` works are intended and that any files that aren't included still end up the __all_files target.

File: internal/npm_install/generate_build_file.ts (hunk 2 above)
Comment (gregmagolan) [reply internal/npm_install/generate_build_file.ts]: Yup. Some refactoring here.

**Pull request merged**